benchmarks/benchmarks/interpolate.py

```diff
@@ -93,16 +93,18 @@ def time_evaluation(self, n_grids, method):
 class Interpolate1d(Benchmark):
     param_names = ['n_samples', 'method']
     params = [
-        [10, 50, 100],
+        [10, 50, 100, 1000, 10000],
         ['linear', 'nearest', 'zero', 'slinear', 'quadratic', 'cubic'],
     ]
 
     def setup(self, n_samples, method):
         self.x = np.arange(n_samples)
         self.y = np.exp(-self.x/3.0)
+        self.interpolator = interpolate.interp1d(self.x, self.y, kind=method)
+        self.xp = np.linspace(self.x[0], self.x[-1], 4*n_samples)
 
     def time_interpolate(self, n_samples, method):
-        interpolate.interp1d(self.x, self.y, kind=method)
+        self.interpolator(self.xp)
```

Comment:
Maybe one benchmark for construction and a second one for evaluation?
The construction overhead may in principle be also interesting to track.